### PR TITLE
token rotation

### PR DESCRIPTION
**Pull request by wmosienski**

### CODE REVIEW

File: database/functions/token/find_by_token.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE OR REPLACE FUNCTION FIND_BY_TOKEN_TOKEN(
```
Comment (Mikalai-Kastsevich-ITechArt) [inline database/functions/token/find_by_token.sql]: What does it mean 'Find By Token Token'

File: src/service/user.service.ts [hunk 2]
```diff
@@ -56,13 +60,34 @@ export class UserService implements IUserService {
...
         return { accessToken, refreshToken };
     }
 
-    async refresh(userRefreshDTO: UserRefreshDTO): Promise<string> {
-        const tokenData = verifyToken(userRefreshDTO.refreshToken);
+    public async logout(refreshToken: string): Promise<void> {
+        await this._userRepository.deleteTokenByToken(refreshToken);
+    }
+
+    async refresh(refreshToken: string): Promise<string> {
```
Comment (Mikalai-Kastsevich-ITechArt) [inline src/service/user.service.ts]: The name of the method is not clear

File: database/functions/token/find_by_token.sql (hunk 1 above)
Comment (wmosienski) [reply database/functions/token/find_by_token.sql]: addressed

File: src/service/user.service.ts (hunk 2 above)
Comment (wmosienski) [reply src/service/user.service.ts]: addressed

File: src/controller/user.controller.ts [hunk 3]
```diff
@@ -58,17 +64,49 @@ export class UserController extends BaseController {
...
+        const cookies = req.cookies;
 
-        res.status(HTTPCodes.success.ok).send(result);
+        if (!cookies?.jwt) {
+            res.status(HTTPCodes.client_errors.unauthorized).send("no jwt cookie");
+
+            return;
+        }
+
+        const accessToken = await this._userService.refreshSession(cookies.jwt);
```
Comment (Mikalai-Kastsevich-ITechArt) [inline src/controller/user.controller.ts:107]: What do you don't return here refresh token as well?

**Pull request merged**